[COL] Fixed Keymapping for Boulder Dash

meveric opened the pull request:
This allows Boulder Dash to work, as well as adding different versions of Boulder Dash.
After the first initial "#" for start you are still asked to press 1 or 2 to start a game.

Commit:
[COL] Fixed Keymapping for Boulder Dash

meveric commented:
added more button mappings for more games... probably keep going when I found more...

OtherCrashOverride commented on smsplusgx-go/main/main.c:
```diff
@@ -699,11 +699,19 @@ void app_main(void)
                     }
 
+                    if (joystick.values[ODROID_INPUT_START] &&
+                        joystick.values[ODROID_INPUT_LEFT])
```
This will cause both # and 1 to pressed at the same time.  It would be preferable to combine the logic with the line above so that only one button is pressed:

```
if (joystick.values[ODROID_INPUT_START])
{
  if( joystick.values[ODROID_INPUT_LEFT])
    coleco.keypad[0] = 1;
  else
    coleco.keypad[0] = 11; // #
}
```

OtherCrashOverride replied:
I mistakenly thought it was a bitmask, but its a single value.  So the latter will overwrite the former instead of pressing both at once.

meveric replied:
even if both are pressed at once (# and 1) it wouldn't affect the game as it only looks for a 1 or 2 at this moment, every other character would be discarded anyway. It's only required once to start the game.

Commit:
[COL] - added support for different versions of Frogger  …
- added support for Fraction Fever (requires "*" to start)
- added support for Dr. Seuss's Fix-Up The Mix-Up Puzzler (requires "*" to shuffle picture - use START + LEFT)
- added support for Decathlon (requires "1" and "*" to start - use START + LEFT for "*")
